### ISSUE TITLE
Confused, is this pessimistic or optimistic locking?

### ISSUE THREAD

**Issue by Lambik** [laravel-enso/versions#5]
Hello,

I'm confused; you have a Versionable and a Versions package, both seem to do pretty much the same thing when I compare the code for the trait.

I don't see how this package does pessimistic locking (which you claim it does in the composer.json), or does it do optimistic as well (as in the readme.md)?

Both projects seem to do a `DB::table($this->getTable())->lock()` and a check on "is the version the same, if no throw an exception".

**Comment by npw1982** [comment]
Yes I was wondering which of these two to use? Versioning seems to have an independent table for storing the version info, whereas for Versions the version is stored in the table for the model itself? Versioning seems to be the more popular package?

**Comment by aocneanu** [comment]
Thank you for your contribution.

As part of the Laravel Enso 9.0.0 release reset, we are closing the current backlog of GitHub issues and pull requests created before April 23, 2026.

Enso 9.0.0 is one of the largest frontend/backend releases in the history of Laravel Enso, bringing the ecosystem up to date with Laravel 13, PHP 8.3+, Vue 3.5, Vite 7, Pinia, a redesigned active Enso UI layer, and refreshed public documentation.

Because of the scale of these changes, older reports and pull requests may no longer apply cleanly to the Enso 9 line. If this issue or pull request is still relevant on top of the current Enso 9 stack, please open a new issue or pull request with updated context.

Thank you for understanding.
